app/helpers/course_summaries_helper.rb

```diff
@@ -0,0 +1,70 @@
+module CourseSummariesHelper
+
+	###
+	# Get JSON data for the table in the form of
+	# { id: student_id, user_name: student.user_name,
+	#   first_name: student.first_name, last_name: student.last_name
+	#   marks: [ // an array of marks for each assignment, null if no grade ] }
+	###
+	def get_table_json_data()
+		all_students = Student.where(type: 'Student');
+		all_assignments = Assignment.all();
+
+		studentList = all_students.map do |student|
+			{
+				:id => student.id,
+				:user_name => student.user_name,
+				:first_name => student.first_name,
+				:last_name => student.last_name,
+				:marks => get_marks_for_all_assignments_for_student(student, all_assignments)
+			}
+		end
+
+		studentList.to_json
+
+	end
+
+	# Get marks for all assignments for a student 
+	def get_marks_for_all_assignments_for_student(student, all_assignments)
+		marks = []
+		all_assignments.each do |assignment|
+			marks[assignment.id-1] = get_mark_for_assignment_and_student(assignment, student)
```

Comment:
Line is too long. [84/80]<br>Use 2 (not 1) spaces for indentation.<br>Surrounding space missing for operator '-'.<br>Tab detected.